### PR TITLE
Added SNV normalization

### PR DESCRIPTION
**Pull request by borondics**
At the request of users from Spain.

Standard Normal Variate is calculated by mean centering the spectra and subsequently dividing each with its standard deviation.

### CODE REVIEW

**Comment by borondics** [comment]
The button "pressed" in the test for the "Normalize by Reference" feature is hardcoded. First, it failed and when I changed it it passes. Can we handle this better in the test?

**Comment by stuart-cls** [comment]
See my minmax implementation for a "solution" to the hardcoded button

**Comment by borondics** [comment]
@stuart-cls if you think this is OK please merge it to the master.

**Comment by stuart-cls** [comment]
Can you add a simple test of the calculation? Also there's a rogue "print" function in one of the files :)

**Comment by markotoplak** [comment]
I have squashed all your commits into one. Next time use rebase instead of merge when updating your branch. Merges within branches make history needlessly complicated (and your merged twice).

File: orangecontrib/spectroscopy/widgets/preprocessors/normalize.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ class NormalizeEditor(BaseEditorOrange):
         ("Min-Max Normalization", Normalize.MinMax),
         ("Area Normalization", Normalize.Area),
         ("Attribute Normalization", Normalize.Attribute),
+        ("Standard Normal Variate (SNV)", Normalize.SNV),
```
Comment (stuart-cls) [inline orangecontrib/spectroscopy/widgets/preprocessors/normalize.py:30]: Is it redundant to say "(SNV)" here? I guess users will know (hopefully) what it is?

**Comment by borondics** [comment]
@stuart-cls I wouldn't count on them knowing it. :)
I didn't know what it means before I looked it up.

**Comment by stuart-cls** [comment]
I just meant having both the name and the acronym :)

**Comment by stuart-cls** [comment]
Okay, good news. We save the Normalization method as the setting, not the radio button position (duh I guess, good job past us :) ) so no problem with the settings.

**Comment by markotoplak** [comment]
@stuart-cls, today I also thought of the same settings problem and then spent an hour looking for how could our settings ever work before the changes. Then I saw that everything is fine, we just add ids to radio buttons. Seems like you went through a similar process. :)

**Comment by borondics** [comment]
Well, we could drop the long name, but then we really have to start writing
documentation for it! :)

**Pull request merged**